src/texture/tex_impl.rs

```diff
@@ -106,7 +107,8 @@ impl TextureImplementation {
         };
 
         let (tx, rx) = channel();
-        display.context.context.exec(move |: ctxt| {
+        //This always syncs with rx later.
+        display.context.context.exec_maybe_sync(false, move |: ctxt| {
```

Comment:
Hmm, what's the point of using `exec_maybe_sync` if you're always passing `false`?